query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -221,11 +221,18 @@ trait ExecPlan extends QueryCommand {
                   MeasurementUnit.time.milliseconds)
               .withTag("plan", getClass.getSimpleName)
               .record(Math.max(0, System.currentTimeMillis - startExecute))
+            // Does not include sizes of RV's that arrived at this node as SerializableRangeVectors.
             val numDataBytes = builder.allContainers.map(_.numBytes).sum
             val numKeyBytes = r.foldLeft(0)(_ + _.key.keySize)
             val resultSize = numDataBytes + numKeyBytes
             SerializedRangeVector.queryResultBytes.record(resultSize)
-            querySession.queryStats.getResultBytesCounter(Nil).addAndGet(resultSize)
+            // Includes sizes of all RV's in this subtree.
+            val totalSize = querySession.queryStats.getResultBytesCounter(Nil).addAndGet(resultSize)
+            if (totalSize > queryContext.plannerParams.resultByteLimit) {
```

Comment:
It would be nice if this check could be done with each RV (i.e. keep a running byte total), but there isn't really a great way to get a `SerializableRangeVector`'s size without iterating through its builder's containers each time, ala:
```
startBytes = builder.allContainers.map(_.numBytes).sum
SerializedRangeVector(builder, other_args)
endBytes = builder.allContainers.map(_.numBytes).sum
srvBytes = endBytes - totalBytes
```

If that is preferable to a single check at the end, then I would also need to count sizes of the RV's that are already SRV's before they're processed. I haven't yet disproven the assumption that "any already-SRV we process must have its size included in the `QuerySession` counter", so I could just check that `counter + sizes_of_other_RVs` doesn't exceed the limit. I'm not sure how contentious the counter lock might become, though.

The other option would be to just give each SRV a `size` attribute and calculate on construction. This would get weird with scalars, though. I could probably just ignore things like fixed scalars, but I would need to handle the "varying" flavors like:
```
final case class ScalarVaryingDouble(private val timeValueMap: Map[Long, Double],
                                     override val outputRange: Option[RvRange]) extends ScalarRangeVector {
```
I could ballpark it?

All of this is mostly to say: I think this single check is good enough.